core/src/main/java/org/bitcoinj/wallet/AnyBasicKeyChain.java

```diff
@@ -0,0 +1,650 @@
+/*
+ * Copyright 2013 Google Inc.
+ * Copyright 2019 Andreas Schildbach
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.wallet;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+import org.bitcoinj.core.BloomFilter;
+import org.bitcoinj.crypto.IKey;
+import org.bitcoinj.core.NetworkParameters;
+import org.bitcoinj.crypto.EncryptableItem;
+import org.bitcoinj.crypto.EncryptedData;
+import org.bitcoinj.crypto.KeyCrypter;
+import org.bitcoinj.crypto.KeyCrypterException;
+import org.bitcoinj.crypto.KeyCrypterScrypt;
+import org.bitcoinj.crypto.factory.KeyFactory;
+import org.bitcoinj.utils.ListenerRegistration;
+import org.bitcoinj.utils.Threading;
+import org.bitcoinj.wallet.listeners.KeyChainEventListener;
+import org.bouncycastle.crypto.params.KeyParameter;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.Executor;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+/**
+ * A {@link KeyChain} that implements the simplest model possible: it can have keys imported into it, and just acts as
+ * a dumb bag of keys. It will, left to its own devices, always return the same key for usage by the wallet, although
+ * it will automatically add one to itself if it's empty or if encryption is requested.
+ */
+public class AnyBasicKeyChain implements IEncryptableKeyChain {
```

Comment:
This is a copy of BasicKeyChain that can handle any type of key.  

In a future PR, this can probably be refactored away and `BasicKeyChain` can handle any kind of key.